### PR TITLE
Throw exception if source zip file doesn't exists

### PR DESCRIPTION
**Pull request by jan-onderka**
Update for better error message during smoke tests handled by PROD team.

### CODE REVIEW

File: core/src/main/groovy/noe/common/utils/InstallerUtils.groovy [hunk 1]
```diff
@@ -80,8 +80,13 @@ class InstallerUtils {
       log.info ("Downloading ${downloadZipUrl} to ${zipDest}")
       Library.downloadFile(downloadZipUrl, zipDest)
     } else {
+      if (!JBFile.isExistingFile(zipSource)) {
+        throw new FileNotFoundException("Expected zip file ${zipSource} doesn't exist.")
+      }
       File dest = new File(basedir)
-      JBFile.copy(zipSource, dest)
+      if (!JBFile.copy(zipSource, dest)) {
+        throw new FileNotFoundException("Something goes wrong with copying file ${zipSource} to the targed directory ${dest}. Check debug log for more information.")
```
Comment (PaulLodge) [inline core/src/main/groovy/noe/common/utils/InstallerUtils.groovy:88]: please change `goes` to `has gone`

Comment (PaulLodge) [reply core/src/main/groovy/noe/common/utils/InstallerUtils.groovy:88]: A part from my comment it looks good to me :smile:

File: core/src/main/groovy/noe/ews/utils/EwsUtils.groovy [hunk 2]
```diff
@@ -125,7 +127,11 @@ class EwsUtils extends InstallerUtils {
     try {
       installJwsZips()
     } catch (FileNotFoundException e) {
-      loadEwsZipNamesInCompatibilityMode()
+      try {
+        loadEwsZipNamesInCompatibilityMode()
+      } catch (FileNotFoundException e) {
+        throw new FileNotFoundException("There were multiple attempts for extracting zip content from various zip compatibility names, but non of them succeed. They don't exists. ${sourceZipTested}")
```
[the remark this answers is not in the document]
Comment (jan-onderka) [reply core/src/main/groovy/noe/ews/utils/EwsUtils.groovy]: Fixed by "update 1" commit

[the remark this answers is not in the document]
Comment (jan-onderka) [reply core/src/main/groovy/noe/ews/utils/EwsUtils.groovy]: Fixed by "update 1" commit

**Comment by jan-onderka** [comment]
@mmadzin @PaulLodge @jstefl @AdamKrajcik I finished my changes, could you take a look, please?

File: core/src/main/groovy/noe/ews/utils/EwsUtils.groovy [hunk 3]
```diff
@@ -129,7 +129,9 @@ class EwsUtils extends InstallerUtils {
     } catch (FileNotFoundException e) {
       try {
         loadEwsZipNamesInCompatibilityMode()
-      } catch (FileNotFoundException e) {
+      } catch (FileNotFoundException ex) {
+        log.debug(e.message)
```
Comment (PaulLodge) [inline core/src/main/groovy/noe/ews/utils/EwsUtils.groovy:133]: this log message should be in the `try` statement above

Comment (mmadzin) [reply core/src/main/groovy/noe/ews/utils/EwsUtils.groovy:133]: I think it's correct. Our concept here breaks some best practices (from my POV), when we use exceptions for controlling the execution (we are not using it for "special unexpected" cases.

**Pull request merged**